### Section 1. problem statement and dataset description

Defining the Problem Statements

Using SMART framework:

1. **Specific**: enhancing player loyalty, achiving stable player count and retention on an online gaming  

2. **Measurable**: stable player count.

3. **Achievable**: Improving player experience by studying how, when, and what players play.

4. **Relevant**: Stable player count can leads to healthy playerbase.

5. **Time-Bound**: achive within next season ( 6 months ).

`Problem statement`:
main goal is to achive stable player count and improving player experience, within the next season 6 months prior so the player base would be much healthier. stable player counts can be studied by how,when, and what players play. 

### Section 2. Data Loading and Lists of libraries that will be used

In [3]:
import pandas as pd
from scipy import stats
import numpy as np
import matplotlib.pyplot as plt
import matplotlib.patches as mpatches


#### Dataset kaggle reading

In [4]:
Online_gaming_df = pd.read_csv("online_gaming_behavior_dataset.csv")
Online_gaming_df

,PlayerID,Age,Gender,Location,GameGenre,PlayTimeHours,InGamePurchases,GameDifficulty,SessionsPerWeek,AvgSessionDurationMinutes,PlayerLevel,AchievementsUnlocked,EngagementLevel
0,9000,43,Male,Other,Strategy,16.271119,0,Medium,6,108,79,25,Medium
1,9001,29,Female,USA,Strategy,5.525961,0,Medium,5,144,11,10,Medium
2,9002,22,Female,USA,Sports,8.223755,0,Easy,16,142,35,41,High
3,9003,35,Male,USA,Action,5.265351,1,Easy,9,85,57,47,Medium
4,9004,33,Male,Europe,Action,15.531945,0,Medium,2,131,95,37,Medium
...,...,...,...,...,...,...,...,...,...,...,...,...,...
40029,49029,32,Male,USA,Strategy,20.619662,0,Easy,4,75,85,14,Medium
40030,49030,44,Female,Other,Simulation,13.539280,0,Hard,19,114,71,27,High
40031,49031,15,Female,USA,RPG,0.240057,1,Easy,10,176,29,1,High
40032,49032,34,Male,USA,Sports,14.017818,1,Medium,3,128,70,10,Medium


#### Data Structures. understanding the data by getting the rows and columns
 From the outputs we can gather that, there are a total of 40034 rows and 13 columns. 

In [5]:
Online_gaming_df.shape

(40034, 13)

In [6]:
Online_gaming_df.columns.to_list()

['PlayerID',
 'Age',
 'Gender',
 'Location',
 'GameGenre',
 'PlayTimeHours',
 'InGamePurchases',
 'GameDifficulty',
 'SessionsPerWeek',
 'AvgSessionDurationMinutes',
 'PlayerLevel',
 'AchievementsUnlocked',
 'EngagementLevel']

#### Checking Data Types

In [7]:
Online_gaming_df.dtypes

PlayerID                       int64
Age                            int64
Gender                        object
Location                      object
GameGenre                     object
PlayTimeHours                float64
InGamePurchases                int64
GameDifficulty                object
SessionsPerWeek                int64
AvgSessionDurationMinutes      int64
PlayerLevel                    int64
AchievementsUnlocked           int64
EngagementLevel               object
dtype: object

In [8]:
Online_gaming_df.head(10)

,PlayerID,Age,Gender,Location,GameGenre,PlayTimeHours,InGamePurchases,GameDifficulty,SessionsPerWeek,AvgSessionDurationMinutes,PlayerLevel,AchievementsUnlocked,EngagementLevel
0,9000,43,Male,Other,Strategy,16.271119,0,Medium,6,108,79,25,Medium
1,9001,29,Female,USA,Strategy,5.525961,0,Medium,5,144,11,10,Medium
2,9002,22,Female,USA,Sports,8.223755,0,Easy,16,142,35,41,High
3,9003,35,Male,USA,Action,5.265351,1,Easy,9,85,57,47,Medium
4,9004,33,Male,Europe,Action,15.531945,0,Medium,2,131,95,37,Medium
5,9005,37,Male,Europe,RPG,20.561855,0,Easy,2,81,74,22,Low
6,9006,25,Male,USA,Action,9.752716,0,Hard,1,50,13,2,Low
7,9007,25,Female,Asia,RPG,4.401729,0,Medium,10,48,27,23,Medium
8,9008,38,Female,Europe,Simulation,18.152733,0,Easy,5,101,23,41,Medium
9,9009,38,Female,Other,Sports,23.942772,0,Easy,13,95,99,36,High


#### Statistics summary. min, max, mean, distributions for numeric coloumns

In [9]:
Online_gaming_df.describe()

,PlayerID,Age,PlayTimeHours,InGamePurchases,SessionsPerWeek,AvgSessionDurationMinutes,PlayerLevel,AchievementsUnlocked
count,40034.000000,40034.000000,40034.000000,40034.000000,40034.000000,40034.000000,40034.000000,40034.000000
mean,29016.500000,31.992531,12.024365,0.200854,9.471774,94.792252,49.655568,24.526477
std,11556.964675,10.043227,6.914638,0.400644,5.763667,49.011375,28.588379,14.430726
min,9000.000000,15.000000,0.000115,0.000000,0.000000,10.000000,1.000000,0.000000
25%,19008.250000,23.000000,6.067501,0.000000,4.000000,52.000000,25.000000,12.000000
50%,29016.500000,32.000000,12.008002,0.000000,9.000000,95.000000,49.000000,25.000000
75%,39024.750000,41.000000,17.963831,0.000000,14.000000,137.000000,74.000000,37.000000
max,49033.000000,49.000000,23.999592,1.000000,19.000000,179.000000,99.000000,49.000000


#### Missing Values

In [10]:
Online_gaming_df.isnull().sum()

PlayerID                     0
Age                          0
Gender                       0
Location                     0
GameGenre                    0
PlayTimeHours                0
InGamePurchases              0
GameDifficulty               0
SessionsPerWeek              0
AvgSessionDurationMinutes    0
PlayerLevel                  0
AchievementsUnlocked         0
EngagementLevel              0
dtype: int64

#### Duplicates

In [11]:
Online_gaming_df.duplicated().sum()

0

#### Unique Value Counts

In [12]:
Online_gaming_df.nunique()

PlayerID                     40034
Age                             35
Gender                           2
Location                         4
GameGenre                        5
PlayTimeHours                40034
InGamePurchases                  2
GameDifficulty                   3
SessionsPerWeek                 20
AvgSessionDurationMinutes      170
PlayerLevel                     99
AchievementsUnlocked            50
EngagementLevel                  3
dtype: int64

### Section 3. Data Cleaning 
Since the dataset doesn't have any
- duplicates
- missing values

on data exploration. we can change object dtype to strings, so it wont be ambiguous

In [13]:
obj_change = ['Gender', 'Location', 'GameGenre', 'GameDifficulty', 'EngagementLevel']
Online_gaming_df[obj_change] = Online_gaming_df[obj_change].astype('string')


In [14]:
Online_gaming_df.dtypes

PlayerID                              int64
Age                                   int64
Gender                       string[python]
Location                     string[python]
GameGenre                    string[python]
PlayTimeHours                       float64
InGamePurchases                       int64
GameDifficulty               string[python]
SessionsPerWeek                       int64
AvgSessionDurationMinutes             int64
PlayerLevel                           int64
AchievementsUnlocked                  int64
EngagementLevel              string[python]
dtype: object

### Section 4. Data Analysis and Exploratory Data Analysis (EDA) for SMART framework and the main problem of case study

##### What factors influence player engagement behavior?
To answer this question, we analyze whether there is a significant difference in player activity between different engagement groups.

in particular, we want to understand :
> Do highly engaged players spend more time playing compared to low-engagement players?


We can gather that there is a variable called 
> EngagementLevel: Indicates the level of player engagement categorized as 'High', 'Medium', or 'Low'.

we can identify what it means and how we will use this variable to answer our factors  

In [15]:
# Group by EngagementLevel and calculate average sessions
engagement_summary = (
    Online_gaming_df.groupby('EngagementLevel')['SessionsPerWeek']
      .mean()
      .reset_index()
      .rename(columns={'SessionsPerWeek': 'AvgSessionsPerWeek'})
      .sort_values(by='AvgSessionsPerWeek')
)

engagement_summary

,EngagementLevel,AvgSessionsPerWeek
1,Low,4.530511
2,Medium,9.553267
0,High,14.254547


The results show that the higher engagement level players have, the more sessions they have per week.

From the results we can confirm that EngagementLevel Variable is a summary label that's represents player activity.

since EngagementLevel Confirms a summary label, we can use it to perform Statistical Significance with the current hypothesis :

>$H_0$: There is no difference in mean playtime between low and high engagement players.

>$H_1$: There is a significant difference in mean playtime between low and high engagement players.

We will use confidence level of 95%. which means p-value = 0.05

In [16]:
# grouping/filtering data using labels based on EngagementLevel
low  = Online_gaming_df[Online_gaming_df['EngagementLevel'] == 'Low']['PlayTimeHours']
high = Online_gaming_df[Online_gaming_df['EngagementLevel'] == 'High']['PlayTimeHours']

In [17]:
print("Low Engagement PlayTime:")
print(low)
print("\nHigh Engagement PlayTime:")
print(high)

Low Engagement PlayTime:
5        20.561855
6         9.752716
13        8.701959
14       17.975200
17       17.272113
           ...    
40018     5.950284
40022     9.085086
40023     2.529218
40027     1.273467
40028     1.020489
Name: PlayTimeHours, Length: 10324, dtype: float64

High Engagement PlayTime:
2         8.223755
9        23.942772
10        4.829916
11        5.535981
12       18.776234
           ...    
40013     9.077397
40020    10.687862
40026    19.825140
40030    13.539280
40031     0.240057
Name: PlayTimeHours, Length: 10336, dtype: float64


In [18]:
# Using Two-Sample T-test
t_stat, p_value = stats.ttest_ind(low, high)

print("T-statistic:", t_stat)
print("P-value:", p_value)

T-statistic: 0.3715643967745735
P-value: 0.710221012114041


In [19]:
alpha = 0.05

print("Mean Playtime (Low):", low.mean())
print("Mean Playtime (High):", high.mean())
print()

if p_value < alpha:
    print("Reject H0 : There is a significant difference in playtime.")
else:
    print("Fail to reject H0 : No significant difference detected.")


Mean Playtime (Low): 12.10491461074908
Mean Playtime (High): 12.069238073121078

Fail to reject H0 : No significant difference detected.


Based on Two Sample T-test. the result shows that the average playtime for low engagement players (12.10 hours) and high engagement players (12.07 hours) is nearly identical. meaning that there is no significant difference in playtime between low and high engagement players.

We can infer from the fact that gamers with low and high levels of engagement spend almost the same amount of time playing. It indicates that a stable player count is already being achieved.

### Section 5. Conclusion



based on the analysis and EDA done, we can conclude these findings :

1. Engagement level reflects session frequency, not playtime
High engagement players average ~14 sessions/week vs ~4.5 for low engagement. However, the Two-Sample T-test (p > 0.05) shows no significant difference in playtime hours between the two groups. meaning all players spend roughly the same total time (~12 hours) regardless of engagement level.
2. sports is the most dominant genre. 
With the highest player count, sports games are the top priority for retention and experience improvement efforts.
3. In-game purchase rates are relatively consistent across all genres. 
No single genre stands out in monetization, suggesting purchase behavior is not driven by genre preference
4. Session frequency is fairly consistent across players. 
The mean, median, and mode of SessionsPerWeek are close to each other, indicating a relatively normal distribution with no strong skew, players are predictable in how often they play.

from the lists above, we can conclude that
The playerbase already shows signs of stability. playtime is consistent across engagement groups. To further improve retention within the next 6 months, the platform should focus on increasing session frequency, particularly for low-engagement players, rather than trying to extend total playtime.